{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cc2b1097-5c37-4963-a5c9-242f4665472a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "from config import RUN_GATK, QC, DOWNLOAD_REF_GENOME, RESULTS_FOLDER, DATA_FOLDER, NORMAL_R1, NORMAL_R2, TUMOR_R1, TUMOR_R2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "28a02e4c-5cc9-47dd-9a42-63ddee2ea9da",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download reference human genome\n",
    "if DOWNLOAD_REF_GENOME:\n",
    "    !wget -O \"$DATA_FOLDER/hg38.fa.gz\" http://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz\n",
    "    !gunzip \"$DATA_FOLDER/hg38.fa.gz\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9945c9a-474d-4351-9ad0-d859e1d40866",
   "metadata": {},
   "source": [
    "# Quality control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1f0df6da-750f-4556-aa48-b0c7c15478b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "zsh:1: no matches found: data/*.fastq.gz\n",
      "null\n",
      "null\n",
      "Started analysis of PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "null\n",
      "null\n",
      "Approx 5% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 10% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 15% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 20% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 25% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 30% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 35% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 40% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 45% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 50% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 55% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 60% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 65% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 70% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 75% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 80% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 85% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 90% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Approx 95% complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Analysis complete for PA221MH-lib09-P19-Norm_S1_L001_R1_001.fastq\n",
      "Started analysis of PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 5% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 10% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 15% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 20% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 25% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 30% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 35% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 40% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 45% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 50% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 55% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 60% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 65% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 70% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 75% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 80% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 85% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 90% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Approx 95% complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Analysis complete for PA221MH-lib09-P19-Norm_S1_L001_R2_001.fastq\n",
      "Started analysis of PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 5% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 10% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 15% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 20% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 25% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 30% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 35% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 40% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 45% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 50% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 55% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 60% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 65% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 70% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 75% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 80% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 85% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 90% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Approx 95% complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Analysis complete for PA220KH-lib09-P19-Tumor_S2_L001_R1_001.fastq\n",
      "Started analysis of PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 5% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 10% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 15% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 20% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 25% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 30% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 35% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 40% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 45% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 50% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 55% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 60% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 65% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 70% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 75% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 80% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 85% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 90% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Approx 95% complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "Analysis complete for PA220KH-lib09-P19-Tumor_S2_L001_R2_001.fastq\n",
      "\n",
      "\u001b[91m///\u001b[0m \u001b]8;id=607852;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ üîç \u001b[2mv1.26\u001b[0m\n",
      "\n",
      "\u001b[34m       file_search\u001b[0m | Search path: /Users/drashti/Developer/test/results\n",
      "\u001b[2K         \u001b[34msearching\u001b[0m | \u001b[90m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m42/42\u001b[0m  2mresults/filtered_tumor_variants.csv\u001b[0m\n",
      "\u001b[?25h\u001b[34m            picard\u001b[0m | Found 2 MarkDuplicates reports\n",
      "\u001b[34m            fastqc\u001b[0m | Found 4 reports\n",
      "\u001b[34m     write_results\u001b[0m | Data        : results/multiqc_report/multiqc_data\n",
      "\u001b[34m     write_results\u001b[0m | Report      : results/multiqc_report/multiqc_report.html\n",
      "\u001b[34m           multiqc\u001b[0m | MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "# Perform quality checks using tools like FastQC and summarize quality metrics (e.g., sequence counts, \n",
    "# per-base quality, read duplication levels).\n",
    "\n",
    "# Decompress if needed\n",
    "!gunzip \"$DATA_FOLDER\"/*.fastq.gz\n",
    "\n",
    "if QC:\n",
    "    # Run FASTQC\n",
    "    # The summary of quality metrics is provided in the pdf report\n",
    "    !fastqc \"$DATA_FOLDER/$NORMAL_R1\" \"$DATA_FOLDER/$NORMAL_R2\" \"$DATA_FOLDER/$TUMOR_R1\" \"$DATA_FOLDER/$TUMOR_R2\" -o \"$RESULTS_FOLDER\"\n",
    "    !multiqc \"$RESULTS_FOLDER\" -o \"$RESULTS_FOLDER/multiqc_report\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c66c869-256d-499e-9ac1-4b86386c6441",
   "metadata": {},
   "source": [
    "# Alignment and mutation calling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6575f220-2a30-40da-b642-0c5fdc941fa3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Align the samples to the human genome using tools like Bowtie2 or BWA.\n",
    "# https://gatk.broadinstitute.org/hc/en-us/articles/360035535912-Data-pre-processing-for-variant-discovery\n",
    "\n",
    "# Identify somatic mutations present in the cancer sample but absent in the normal tissue.\n",
    "# https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels\n",
    "if RUN_GATK:\n",
    "    !chmod +x somatic_mutations.sh\n",
    "    !./somatic_mutations.sh \"$DATA_FOLDER/$NORMAL_R1\" \"$DATA_FOLDER/$NORMAL_R2\" \"$DATA_FOLDER/$TUMOR_R1\" \"$DATA_FOLDER/$TUMOR_R2\" \"$DATA_FOLDER/hg38.fa\" \"$RESULTS_FOLDER\" \"$DATA_FOLDER\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "db5ff822-b847-4aff-8364-576541f5cae0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gunzip: can't stat: results/filtered_variants.vcf.gz (results/filtered_variants.vcf.gz.gz): No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!gunzip \"$RESULTS_FOLDER/filtered_variants.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bb0d1e6d-27a6-4955-adaa-ac727e2b64f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "def vcf_to_dataframe(vcf_file_path):\n",
    "    # Read a vcf file as a pandas dataframe\n",
    "    \n",
    "    # Dictionary to store INFO ID and Description\n",
    "    info_dict = {}\n",
    "    \n",
    "    # Read the file\n",
    "    with open(vcf_file_path, 'r') as vcf_file:\n",
    "        lines = []\n",
    "        for line in vcf_file:\n",
    "            if line.startswith(\"##INFO\"):\n",
    "                # Extract the ID and Description using regex\n",
    "                match = re.search(r'ID=(.*?),.*?Description=\"(.*?)\"', line)\n",
    "                if match:\n",
    "                    info_id = match.group(1)\n",
    "                    description = match.group(2)\n",
    "                    info_dict[info_id] = description\n",
    "            elif not line.startswith(\"##\"):  # Skip other metadata lines\n",
    "                lines.append(line.strip())\n",
    "\n",
    "    # Extract the column names from the first line and data from the rest\n",
    "    column_names = lines[0].lstrip(\"#\").split(\"\\t\")\n",
    "    data = [line.split(\"\\t\") for line in lines[1:]]\n",
    "    \n",
    "    # Create a Pandas DataFrame\n",
    "    df = pd.DataFrame(data, columns=column_names)\n",
    "                      \n",
    "    # Split INFO column into separate columns\n",
    "    info_df = split_info_column(df[\"INFO\"])\n",
    "    df = pd.concat([df.drop(columns=[\"INFO\"]), info_df], axis=1)\n",
    "                      \n",
    "    # Split FORMAT columns for 'normal' and 'tumor'\n",
    "    format_fields = df[\"FORMAT\"].iloc[0].split(\":\")\n",
    "    for sample in [\"normal\", \"tumor\"]:\n",
    "        sample_data = df[sample].str.split(\":\", expand=True)\n",
    "        sample_data.columns = [f\"{sample}_{field}\" for field in format_fields]\n",
    "        df = pd.concat([df, sample_data], axis=1)\n",
    "\n",
    "    # Drop original FORMAT and sample columns\n",
    "    df = df.drop(columns=[\"FORMAT\", \"normal\", \"tumor\"])\n",
    "\n",
    "    return df, info_dict\n",
    "\n",
    "# Expand INFO column\n",
    "def split_info_column(info_series):\n",
    "    # Expand INFO column into key-value dictionary rows\n",
    "    info_dicts = info_series.str.split(\";\", expand=False).apply(\n",
    "        lambda x: {item.split(\"=\", 1)[0]: item.split(\"=\", 1)[1] if \"=\" in item else np.nan for item in x}\n",
    "    )\n",
    "    # Convert list of dictionaries into a DataFrame\n",
    "    return pd.DataFrame(info_dicts.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a56e0570-ac3f-4657-9a2c-ccdbc6afe730",
   "metadata": {},
   "outputs": [],
   "source": [
    "VCF_FILE = f\"{RESULTS_FOLDER}/filtered_variants.vcf\"\n",
    "df, info = vcf_to_dataframe(VCF_FILE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4e754a0d-0192-4d47-b840-9965128c721a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AS_FilterStatus': 'Filter status for each allele, as assessed by ApplyVQSR. Note that the VCF filter field will reflect the most lenient/sensitive status across all alleles.',\n",
       " 'AS_SB_TABLE': 'Allele-specific forward/reverse read counts for strand bias tests. Includes the reference and alleles separated by |.',\n",
       " 'AS_UNIQ_ALT_READ_COUNT': 'Number of reads with unique start and mate end positions for each alt at a variant site',\n",
       " 'CONTQ': 'Phred-scaled qualities that alt allele are not due to contamination',\n",
       " 'DP': 'Approximate read depth; some reads may have been filtered',\n",
       " 'ECNT': 'Number of potential somatic events in the assembly region',\n",
       " 'ECNTH': 'Number of somatic events in best supporting haplotype for each alt allele',\n",
       " 'GERMQ': 'Phred-scaled quality that alt alleles are not germline variants',\n",
       " 'MBQ': 'median base quality by allele',\n",
       " 'MFRL': 'median fragment length by allele',\n",
       " 'MMQ': 'median mapping quality by allele',\n",
       " 'MPOS': 'median distance from end of read',\n",
       " 'NALOD': 'Log 10 odds of artifact in normal with same allele fraction as tumor',\n",
       " 'NCount': 'Count of N bases in the pileup',\n",
       " 'NLOD': 'Normal log 10 likelihood ratio of diploid het or hom alt genotypes',\n",
       " 'OCM': \"Number of alt reads whose original alignment doesn't match the current contig.\",\n",
       " 'PON': 'site found in panel of normals',\n",
       " 'POPAF': 'negative log 10 population allele frequencies of alt alleles',\n",
       " 'ROQ': 'Phred-scaled qualities that alt allele are not due to read orientation artifact',\n",
       " 'RPA': 'Number of times tandem repeat unit is repeated, for each allele (including reference)',\n",
       " 'RU': 'Tandem repeat unit (bases)',\n",
       " 'SEQQ': 'Phred-scaled quality that alt alleles are not sequencing errors',\n",
       " 'STR': 'Variant is a short tandem repeat',\n",
       " 'STRANDQ': 'Phred-scaled quality of strand bias artifact',\n",
       " 'STRQ': 'Phred-scaled quality that alt alleles in STRs are not polymerase slippage errors',\n",
       " 'TLOD': 'Log 10 likelihood ratio score of variant existing versus not existing'}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "51a6625d-f5d0-479f-b491-e663110a310c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'AS_FilterStatus',\n",
       "       'AS_SB_TABLE', 'DP', 'ECNT', 'ECNTH', 'GERMQ', 'MBQ', 'MFRL', 'MMQ',\n",
       "       'MPOS', 'NALOD', 'NLOD', 'POPAF', 'TLOD', 'RPA', 'RU', 'STR', 'STRQ',\n",
       "       'normal_GT', 'normal_AD', 'normal_AF', 'normal_DP', 'normal_F1R2',\n",
       "       'normal_F2R1', 'normal_FAD', 'normal_PGT', 'normal_PID', 'normal_PS',\n",
       "       'normal_SB', 'tumor_GT', 'tumor_AD', 'tumor_AF', 'tumor_DP',\n",
       "       'tumor_F1R2', 'tumor_F2R1', 'tumor_FAD', 'tumor_PGT', 'tumor_PID',\n",
       "       'tumor_PS', 'tumor_SB'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "15cbd5b1-e719-43ad-bbea-178c7076b4dc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:>"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABHcAAACMCAYAAAD2tUXSAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8qNh9FAAAACXBIWXMAAAsTAAALEwEAmpwYAAAc50lEQVR4nO3dfbRddX3n8fcXwoMkPDiCUQENVNFBQ1Ei1PowQapFsaLVrlofOthxMlRxtKIW6rSLzozjs2Khq44LRWeKBqU41odR7JhYnY5ogglPgRYxM0BRQAQMoBj5zh+/3yU7J/uee+7N3efcnbxfa92Vc/be55zP2fu3f799vmefnchMJEmSJEmS1E97TDqAJEmSJEmS5s7ijiRJkiRJUo9Z3JEkSZIkSeoxizuSJEmSJEk9ZnFHkiRJkiSpxyzuSJIkSZIk9diiLp704IMPzmXLlnXx1GN17733snjx4knHAMwyHbO0M0s7s7QzSzuztDNLO7O0M0s7s7QzSzuztDNLO7O0W0hZdtb69evvyMxDdpiRmTP+AQcBlwDXAZuAZwxb/rjjjstdwZo1ayYd4SFmaWeWdmZpZ5Z2ZmlnlnZmaWeWdmZpZ5Z2ZmlnlnZmaWeWdgspy84C1mVLHWbUM3c+DHwlM18eEXsD+81HxUmSJEmSJEk7Z8biTkQcCDwHOA0gMx8AHug2liRJkiRJkkYxygWVjwBuBy6MiO9FxAURsWv8WE2SJEmSJKnnovxka8gCESuAbwPPzMzLI+LDwD2Z+acDy60CVgEsXbr0uNWrV3cUeXxuu/NufnT/pFMUSx+GWarlhx740O0tW7awZMmSyYVpMEs7s7QbR5arbrl7pOUmvU837epZmv3XbOxubXdUZmlnlnZmaWeWdmZpZ5Z2Zmlnlm6ceOKJ6zNzxeD0UYo7jwK+nZnL6v1nA2dl5inTPWbFihW5bt26nUu8AJx30ef5wFWd/Idis3bm8q1mqTa/e1vTW7t2LStXrpxYliaztDNLu3FkWXbWl0ZabtL7dNOunqXZf83G7tZ2R2WWdmZpZ5Z2ZmlnlnZmaWeWdmbpRkS0Fndm/FlWZv4QuCkinlgnnQRcO8/5JEmSJEmSNAejfiX5RuCi+j9l3Qi8trtIkiRJkiRJGtVIxZ3M3ADscNqPJEmSJEmSJmuU/y1LkiRJkiRJC5TFHUmSJEmSpB6zuCNJkiRJktRjFnckSZIkSZJ6zOKOJEmSJElSj1nckSRJkiRJ6jGLO5IkSZIkST1mcUeSJEmSJKnHLO5IkiRJkiT1mMUdSZIkSZKkHrO4I0mSJEmS1GMWdyRJkiRJknrM4o4kSZIkSVKPLZp0AGm2lp31pYdun7l8K6c17k/S7pxl87tPGdtrSdJ8WzYP/eXuPAYMM9csjiuSJM2OZ+5IkiRJkiT1mMUdSZIkSZKkHrO4I0mSJEmS1GMWdyRJkiRJknrM4o4kSZIkSVKPWdyRJEmSJEnqMYs7kiRJkiRJPWZxR5IkSZIkqccs7kiSJEmSJPWYxR1JkiRJkqQes7gjSZIkSZLUYxZ3JEmSJEmSemzk4k5E7BkR34uIL3YZSJIkSZIkSaObzZk7bwI2dRVEkiRJkiRJszdScSciDgNOAS7oNo4kSZIkSZJmY9Qzd84F3g482F0USZIkSZIkzVZk5vAFIl4EvDAzXx8RK4G3ZuaLWpZbBawCWLp06XGrV6+e/7Rjdtudd/Oj+yedolj6MMzSwiztxp1l+aEHTjtvy5YtLFmyZHxhhtjdslx1y90jLbc7t91husgybF8ZZndru6Oaryyj7ivD7Optd67mmmWu+8owu2LbnQ+jZpmP/WQmu0Lb7YJZ2i3ULF30X7PRx/5lHBZSlp114oknrs/MFYPTRynuvAt4DbAV2Bc4ALg0M1893WNWrFiR69at27nEC8B5F32eD1y1aNIxADhz+VaztDBLu3Fn2fzuU6adt3btWlauXDm2LMPsblmWnfWlkZbbndvuMF1kGbavDLO7td1RzVeWUfeVYXb1tjtXc80y131lmF2x7c6HUbPMx34yk12h7XbBLO0WapYu+q/Z6GP/Mg4LKcvOiojW4s6MP8vKzLMz87DMXAa8Avj6sMKOJEmSJEmSxmc2/1uWJEmSJEmSFphZnceWmWuBtZ0kkSRJkiRJ0qx55o4kSZIkSVKPWdyRJEmSJEnqMYs7kiRJkiRJPWZxR5IkSZIkqccs7kiSJEmSJPWYxR1JkiRJkqQes7gjSZIkSZLUYxZ3JEmSJEmSeszijiRJkiRJUo9Z3JEkSZIkSeoxizuSJEmSJEk9ZnFHkiRJkiSpxxZNOoCk/lt21pemnXfm8q2cNmT+OJlFkzZsXxlmIbUXs2gc5rqvDLOQ2otZpF1XF/3XbCykfXohZfnEyYsnHaFznrkjSZIkSZLUYxZ3JEmSJEmSeszijiRJkiRJUo9Z3JEkSZIkSeoxizuSJEmSJEk9ZnFHkiRJkiSpxyzuSJIkSZIk9ZjFHUmSJEmSpB6zuCNJkiRJktRjFnckSZIkSZJ6zOKOJEmSJElSj1nckSRJkiRJ6rEZizsRcXhErImIayPimoh40ziCSZIkSZIkaWaLRlhmK3BmZl4REfsD6yPia5l5bcfZJEmSJEmSNIMZz9zJzFsz84p6+6fAJuDQroNJkiRJkiRpZrO65k5ELAOeClzeSRpJkiRJkiTNSmTmaAtGLAG+AbwzMy9tmb8KWAWwdOnS41avXj2fOSfitjvv5kf3TzpFsfRhmKWFWdqZpZ1Z2pmlnVnamaWdWdqZpZ1Z2pmlnVnamaWdWdodceCeLFmyZNIx5sWJJ564PjNXDE4fqbgTEXsBXwS+mpkfnGn5FStW5Lp16+YUdCE576LP84GrRrksUffOXL7VLC3M0s4s7czSziztzNLOLO3M0s4s7czSziztzNLOLO3M0u4TJy9m5cqVk44xLyKitbgzyv+WFcDHgE2jFHYkSZIkSZI0PqNcc+eZwGuA50bEhvr3wo5zSZIkSZIkaQQzniOVmd8CYgxZJEmSJEmSNEuz+t+yJEmSJEmStLBY3JEkSZIkSeoxizuSJEmSJEk9ZnFHkiRJkiSpxyzuSJIkSZIk9ZjFHUmSJEmSpB6zuCNJkiRJktRjFnckSZIkSZJ6zOKOJEmSJElSj1nckSRJkiRJ6jGLO5IkSZIkST1mcUeSJEmSJKnHLO5IkiRJkiT1mMUdSZIkSZKkHrO4I0mSJEmS1GMWdyRJkiRJknrM4o4kSZIkSVKPWdyRJEmSJEnqMYs7kiRJkiRJPWZxR5IkSZIkqccs7kiSJEmSJPWYxR1JkiRJkqQes7gjSZIkSZLUYxZ3JEmSJEmSeszijiRJkiRJUo9Z3JEkSZIkSeqxkYo7EXFyRFwfETdExFldh5IkSZIkSdJoZizuRMSewF8CLwCOBn4vIo7uOpgkSZIkSZJmNsqZO8cDN2TmjZn5ALAaOLXbWJIkSZIkSRrFKMWdQ4GbGvdvrtMkSZIkSZI0YZGZwxeIeDlwcma+rt5/DXBCZp4xsNwqYFW9+0Tg+vmPO3YHA3dMOkRllnZmaWeWdmZpZ5Z2ZmlnlnZmaWeWdmZpZ5Z2ZmlnlnZmabeQsuysx2XmIYMTF43wwFuAwxv3D6vTtpOZHwU+Oud4C1BErMvMFZPOAWaZjlnamaWdWdqZpZ1Z2pmlnVnamaWdWdqZpZ1Z2pmlnVnaLaQsXRnlZ1nfBZ4QEUdExN7AK4C/7TaWJEmSJEmSRjHjmTuZuTUizgC+CuwJfDwzr+k8mSRJkiRJkmY0ys+yyMwvA1/uOMtCtJB+ZmaWdmZpZ5Z2ZmlnlnZmaWeWdmZpZ5Z2ZmlnlnZmaWeWdmZpt5CydGLGCypLkiRJkiRp4RrlmjuSJEmSJElaoHar4k5EnBwR10fEDRFxVsv8fSLi4jr/8ohYNjD/sRGxJSLeWu/vGxHfiYiNEXFNRPx511ki4viI2FD/NkbESxuPeVNEXF2zvHkMWfaKiE9GxFURsSkizp7genlVY71siIgHI+LYOm9tfc6peY/sOEvrNprQelkWEfc38nyk8ZjfjYgra5b3jCHLIyJiTd2Hzh94zFSWWyPijtqOPx0R+w4sd1pE3N54P68bmH9ARNw89fwRsf9Au7gjIs4d8X0eFBGXRMT36zq8ub6vvev8D0bEL+u8LRFxXuOxm+t+sSEi1rU895kRkRFx8IhZ/qhup00R8cO6br8WEQ+v88+uObLmfdYM62W/iPhSRFxXn/fdo+QYeM4j6va9YWC9nN5479+KiFUdtJevNPajj0TEniNm7qLtvjMiboqILaOuu53MMmwMGNru5jtLY/52Y2Nj+p4R8b2I+OKkskTE4XXbXVvby5u6zjLDNvp4RNwWpa+bdryOiAMj4guNdv7agfd4WZT+4NrG67buk7N4v62Pj3napwfyb9deYlsf19r3T5N3pu3znIi4IiK2RsTLG9MfV6dvqK95emPeWMfFOu+YiPg/9TWvmnrvMf6+5XkRsb5mWB8Rz208ZtzHUXtHxIU1y8aIWNl4zILpc+v8sfVzw7LEtmOW66L0Dc/oOEvr8WXs3HHXvI9FMeZxMYZ/HlkQbTcmMC425g+O0U8cWF8/izI+Tjc2PilKf/nzge087XuKiPfV/eLKiPhcRBw06vsd9p6G7Y8TlZm7xR/lYtDfB44E9gY2AkcPLPN64CP19iuAiwfmXwJ8FnhrvR/Aknp7L+By4Ne6zALsByyqtx8N3Ea5dtJTgKun5gN/Bzy+4yyvBFY3cm0Glk1ivQwssxz4fuP+WmDFuNrLkG00ifayDLi65TkfAfw/4JB6/5PASR1nWQw8CzgdOL8ly3LgB8BfAycBnwFOG3ju05qPbcn3YeBT0y0DrAeeM2Ib+CTwuprjVcBBwEeAP6zzXwh8sd5+MfCVxmM3AwdP87yHUy5Q/3+nW2Zg+UPrenkY8F7gyroezgLeU5c5GjgG+G/AW4Drhq2X2kZPrLf3Br4JvGCW+8hngFfU2831ckBjmZcA981ne2m+BmWf+pupHONuu3Xer1H28y2zWHfz3r/M1O66yNKYv93Y2Jj+ltruvjipLHUdPa3e3h/4x8HnHPM2eg7wO8DPGDJeA3/Ctv37EOBOYO96fy3wvHp7CbDfsH1yoezTQ7bRQ31cI8dpM2QdZfssY1u/+PLG9L2BfRrrbzPwGCYzLi6i9Om/Wu8/Atiz3h533/JU4DH19lOAWxqPWct4j6PeAFxYbz+SMnbvMaH1Mu3+XKeNrZ8blqW219c12vhBHWdZRsvxZctrjHTctTNZGvN3GIuY0LhYpw9+HlkQbZcJjIvDtlFj3jHAL4AnMv3Y+Ejg6cA7B7bztO8JeH5jPbyHOrbOYhvMqW+Y1N/udObO8cANmXljZj4ArAZOHVjmVErnCKXxnRQRARARL6EcfDz0P4VlMVV93av+ZZdZMvO+zNxap+/beL1/CVzemP8N4Le7zFJfe3FELKJ88HwAuGcS62Vgmd+rj90Z876NFsB6aToS+KfMvL3e/zvgZV1mycx7M/NblA81O2ShfHhZBHwLeDml0/znETIBEBHHAUuBy6aZfxRlUPhmvX9IRPxNRHy3/j2zseyBlA9hHwOeS+nI76rv6yV1sfsaT7+Y0bYlwIeAtzeXj4jFUb7R/06UbwAH1+nUPnYq8CPKenkoS2Zem5lXAg+yfb/Qul5qG11Tbz8AXAEcNtN6aTxn1PVySZ3UzHJPY9FjgHvnub00X2MR5QCj6353WJZvZ+atI7z+fGWZbgyYq3kfG+v0w4BTgAsmmSUzb83MK+rtnwKbKMWEzrIM20aZ+feUYs19M4zXCexf39sSSv+4NSKOphw8fq0+35bMvG/YPrlQ9un6Oi+hpb1Q+7h6PDFK3z/j9snMzY1+sTn9gcz8eb27D9vOYh/7uEj50HFlZm6s2X6cmb+st8fdt3wvM6fW+zWU7bHPLF9/XrJQvrD4OkBm3gbcBayo9xdMnzvufm66LAPHLFNt/K4us4zyRgePu7rMMqRvmYtOPo8slLY7iXERRtpGvwPcmZnXTzc2ZuZtmfldShGoOX3a95SZlzXWw7fZdry7Zz2r57v1rJ5/N02ucR2PzYvdqbhzKHBT4/7N7NiQH1qmbqy7gUdExBLgj4EdfkZTG8YGSrXua5l5eZdZ6mueEBHXAFcBp9f5VwPPjvLzgf0oZxUc3nGWS4B7gVsp33a9PzPvrBnHvl4afhf49MC0C+tpc3864qDUxTaa1Ho5ohYMvhERz67TbgCeGOW02kWUg/iu28t0bqBU6fcCPgicD/wBcHdmthVqXlY74Usi4nCAiNgD+ADw1pblp0xV26c63w8DH8rMp1MO4JsHZ0cAt1O+jdsf+EhELG55vysj4ufAx4G/aExP4LIop7avmppYiza3TB3IN7wD+HpmHg+cCLyvvh6ZeQvwfso+9gTgtrpefkgp2kw990uBlwJnU9bfSOslyumpvwX8rxHWy5RHAHc1BrXt1ktEvCEivg/8e2BN43Hz0V6mXuOrlP3op2z7QDpMF213rjrpX5im3XWVZdjYCJxLKWI+2DJv3FmA8jMCytkJkxqnp1xP+WJk2Hh9PuVLm3+uz/GmzHwQOAq4KyIurf36+6L8LHHYPrkg9unpttFAH3cr0/f9ra8xJMe0opzCf2V9jvfUosYkxsWjgIyIr0b5qdjbR30PHWRpehlwRaMIBuM9jtoIvDgiFkXEEcBxjLYtusgybH8+lzH2c0OyTB2zXFj7hQumjiG6zEL78WXT4HFXJ1lm6P/HOi4OLNP2eWS2uhyLxjYujjJGU4q6MYfPstuZ4T39AfA/6+1/Qxlvnk45G+jf1v5m0JzX7yTsTsWdnXEO5cBoh99IZuYvM/NYShXw+Ih4StdhMvPyzHwypSGeHRH7ZuYmyqlmlwFfATYAv+w4yvH1NR5DGVjOjIgja8axrxcoOxnlG9GrG5NflZnLgWfXv9d0naNtG9Xp414vtwKPzcynUk8djogDMvMnwB8CF1O+UdlM9+2lVSPLJZR97aOUNrw4Il49sPgXgGWZeQzwNbZV0l8PfDkzbx7yUq9g+0H2N4Dza7Htb4ED6uAD5VvkpwEXUjr0eyk/g2q6AnhUZu5DKexc3Jj3rMx8GvAC4A1Rrv2wH+WnFn/Wku35wFk1y1rKNwCPBYhyXZ1TKfvYPdT1Ug+WmmcEfA74HOWnW/9plPVSP8B8GviLzLxxhPUyksz8y8z8FcpP7I6dzWNn8Rq/STkNdh/KGQe7jen6F1raXcdRzqFlbIyIF1GKkOs7fv0ZszQyLaH8hO/Nuf2ZKJ0Yso2gnNZ+B8PH69+s0x9D2YfOj4gDKH3TsykF26dTzjY5bYY4C2WfPof29tLs4x5De98/rzLzpjqOPB741xGxdELj4iLKTz5fVf99aUSc1PFrDhURT6YcTza/xR73cdTHKR8c11EKKP/AhI5RoH1/nlA/N13fMnXM8lf1eK/tmGW+tR5fDiwzeNzVlXOYvv8f97gITPt5ZOyGjUVjHhfPYfgYvTfwr4B3sROfZYe9p4h4B7AVuKhOej7w+3VsvJxSsHnCbF5vhrF+Inan4s4tbF/9O6xOa12mfug5EPgxcALw3ojYDLwZ+JOIOKP5wCynP64BTu44S/M1NwFbKL+NJjM/lpnHZeZzgJ9Qfm/YZZZXUq4z8ossp83+b+pps42MdzHe9bLDQFK/FZw6Te9TlKLUOLLssI0a0+9iDOslM3+emT+ur7me8qHiqHr/C5l5QmY+g/JNctftZVqZ+QXKweQllM78euBS4NcHlvtx45vECyjf5gE8Azij7qPvp3TWD10kOCJ+lfJThuZB2B6Uax4dW/8Ozcwt9YyQTwD3UwpIB1GKJk9rvt/MvKcxSJ0NLIl6geRGm7utPvZ44FcoH1421pyHAVdExKMo1455WSPLYzNzU0RcSCkiPSHLTwV+RDnD5tcjYur3vYM2AUfWLEPXC6WQ9k+Zee4o66V+a3sBZXseVLcvtLcDKNfOaH4LMi/tZUpm/gz4PDueFtymk7Y7R12NAW3trsss042Nz6R8676Zctr2cyPiryeUhYjYi3Kwd1FmXjpCjp3N8pDpxgDgJzOM168FLs3iBsqp7E+ifODdkOW0+K3A/6D0TcP2yYWyT0+3jX4D+EFm3p6Zv6Cl728xyvaZUZYzdq6mFCwmMS7eDPx9Zt6RmfcBX6Zsz7naqXYb5WdGnwN+PzO/P/WAcR9HZebWzPyj2l5PpYzDo2yLec/SXGBgf55EPzddlpuBm3Pb2eCXMFo76uT4si7bdtzVSRaG9P8TGBenzFdhq5OxaALj4kyfpV9AOVvwQ3P4LMtM7ykiTgNeRClUT30xGsAbG2PjEZl5WZQLX2+oRZ9h7+khQ8b68csJX/RnXH+UqvaNlIOSqYtAPXlgmTew/QWTPtPyPOew7SKAh1AvWEa5HsY3gRd1maU+ZuriTY+jnLJ9cL3/yPrvY4HrGO1iajuT5Y/ZdsG7xcC1lN/jj3291Pt7UHbAIweec2r97EUZ8E6fxDaaUHs5hG0XZzyyrp9/MdBeHk4pqBw1jv2IlosiU36TfQKlMLGRcoDwSUqn21zu0Y3bLwW+3ZKx7fnfDfz5wLRPAW9r3D92YP43KT8X+2xtN++jXGT09XX+M4Cot98C/JwyUCwG9m/sF/8AnNySc3Ojbf4Xys8wpp7vqY3lTqD8Pnm/mmE98EbKt3Lvrcs8vr72J4C31e0cw9YL8J8pg+Aes1kvjemfZfuLr06tlyc0ljm1rpd5ay+U6488utEeLwbOmFTbbcybzQUSu+hfRmp3XayXOv0c2i+QuJLRLzTaxTgdlAvqnjvq9ulqGzUetwzYVG+3jtfAXwHn1NtLKfv0wZQLWm5k20V/LwTeMMM+uaD26ZZt1Ozjgpa+fy7bp7HsJ9j+gsqHse3izQ+nfHhYXu+PdVysr3MF219c+5SBx46rbzmoLv/bLc857uOo/YDF9fbzKAWwBdXnDjx2JWPo54ZloR6zNPav93WcZdrjyzpth+OurrIMLHMO2/qWiYyLtHweWUhtlwmMi9Nto8a01ZQvNUb6LDv4HMPeE+WL9Gup42Zj+irKFyR71ftHUfudnd0fJ/k30Rcf+5stv937R0p1+R112n8EXlxv70s5uLkB+E7bTjnQaRwDfI/yPx1cDfxZ11kop8JeQznouAJ4SeM5v1kb70ZG+B8e5iHLkjr9mvq6b5vUeqnzVjLwYZ/Sma+vWa6hXHtgz0lsowm1l5cNZPmtxnN+um63axnhfxuap220mXJh0C2Ub5qOHshyO2VAvBr475Sf3DSf+131/WyknPn0pJZ8p7FjcefGwWUpA9zFdXtcS+24G/OPpZwSfh3l24Mba1s6o85fQ7nA7v2U39++qk4/subbWLO+Y5r1uJltB2UPA/4r5Te71zBwkEj5jfJ1lOLXrXXd3gi8ss4/j3JxuV/Wf38wbL1QPtxkfb4N9e91o6yXxvMdWbfvDXV7T/3vMx9mW5tbQ/kfpuatvVA+6H6XbfvReYz4vxPQTdt9b73/YP33nAn1LyO1u/leL43nOIedLO50kYXyU5es7WVD/XvhJLZRo6+7tbaXX1B+9nlSnXc69UMz5edJl1H6hKuBVzee43n1/VxFKVxM/S9a0+2TC2afnq69sK2Pe6jvn4ft83TKPnkv5VvWawbW38b676qB7TPucfHVdf1eTS3YT6hv+Q91XW1o/D2SyRxHLaOcObWJUvB63ATXy7T787j7uWFZ2HbMciXlA+vDO84y7fFlnb/DcVeX+1Fb38KExkVaPo8spLbLBMbF6bZRvb+Y0kcfSMtnWbYfGx9V1909lAut3wwcMOw91Sw3NaZPFWr2oHy5OjXOrgEObMk7575hEn9T3xJLkiRJkiSph3ana+5IkiRJkiTtcizuSJIkSZIk9ZjFHUmSJEmSpB6zuCNJkiRJktRjFnckSZIkSZJ6zOKOJEmSJElSj1nckSRJkiRJ6jGLO5IkSZIkST32/wGs5D1G1UvU+gAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 1440x144 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "df[\"normal_AF\"].hist(figsize=(20,2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a21a4f19-5e01-4485-9085-999667670941",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AS_FilterStatus</th>\n",
       "      <th>AS_SB_TABLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>...</th>\n",
       "      <th>tumor_AD</th>\n",
       "      <th>tumor_AF</th>\n",
       "      <th>tumor_DP</th>\n",
       "      <th>tumor_F1R2</th>\n",
       "      <th>tumor_F2R1</th>\n",
       "      <th>tumor_FAD</th>\n",
       "      <th>tumor_PGT</th>\n",
       "      <th>tumor_PID</th>\n",
       "      <th>tumor_PS</th>\n",
       "      <th>tumor_SB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows √ó 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CHROM, POS, ID, REF, ALT, QUAL, FILTER, AS_FilterStatus, AS_SB_TABLE, DP, ECNT, ECNTH, GERMQ, MBQ, MFRL, MMQ, MPOS, NALOD, NLOD, POPAF, TLOD, RPA, RU, STR, STRQ, normal_GT, normal_AD, normal_AF, normal_DP, normal_F1R2, normal_F2R1, normal_FAD, normal_PGT, normal_PID, normal_PS, normal_SB, tumor_GT, tumor_AD, tumor_AF, tumor_DP, tumor_F1R2, tumor_F2R1, tumor_FAD, tumor_PGT, tumor_PID, tumor_PS, tumor_SB]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 47 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df[\"normal_AF\"].isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a8b6b8d9-a1b4-4598-93e8-45696eab093c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.039"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Use the normal tissue to calculate the median background mutation level.\n",
    "# The background mutation levels accounts for sequencing errors or biases\n",
    "# that can mimic true mutations.\n",
    "df['normal_AF'] = df['normal_AF'].astype(float)\n",
    "df['tumor_AF'] = df['tumor_AF'].astype(float)\n",
    "normal_af = df[\"normal_AF\"].median()\n",
    "normal_af"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "cf66feea-7298-4baa-94b0-59a8bbe6c594",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39000.0"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Determine how many reads per million are required to confidently call a given mutation.\n",
    "normal_rpm = normal_af * 1_000_000\n",
    "normal_rpm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "28ce99d5-53a4-4682-8405-32c5fbb500ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>AS_FilterStatus</th>\n",
       "      <th>AS_SB_TABLE</th>\n",
       "      <th>DP</th>\n",
       "      <th>...</th>\n",
       "      <th>tumor_AD</th>\n",
       "      <th>tumor_AF</th>\n",
       "      <th>tumor_DP</th>\n",
       "      <th>tumor_F1R2</th>\n",
       "      <th>tumor_F2R1</th>\n",
       "      <th>tumor_FAD</th>\n",
       "      <th>tumor_PGT</th>\n",
       "      <th>tumor_PID</th>\n",
       "      <th>tumor_PS</th>\n",
       "      <th>tumor_SB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26773694</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>104,453|1,21</td>\n",
       "      <td>581</td>\n",
       "      <td>...</td>\n",
       "      <td>276,12</td>\n",
       "      <td>0.050</td>\n",
       "      <td>288</td>\n",
       "      <td>97,8</td>\n",
       "      <td>137,3</td>\n",
       "      <td>245,12</td>\n",
       "      <td>0|1</td>\n",
       "      <td>26773694_G_C</td>\n",
       "      <td>26773694</td>\n",
       "      <td>45,231,1,11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26773703</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>131,481|1,21</td>\n",
       "      <td>634</td>\n",
       "      <td>...</td>\n",
       "      <td>299,12</td>\n",
       "      <td>0.047</td>\n",
       "      <td>311</td>\n",
       "      <td>115,9</td>\n",
       "      <td>146,2</td>\n",
       "      <td>264,12</td>\n",
       "      <td>0|1</td>\n",
       "      <td>26773694_G_C</td>\n",
       "      <td>26773694</td>\n",
       "      <td>57,242,1,11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26773706</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>133,481|1,21</td>\n",
       "      <td>644</td>\n",
       "      <td>...</td>\n",
       "      <td>299,12</td>\n",
       "      <td>0.047</td>\n",
       "      <td>311</td>\n",
       "      <td>114,9</td>\n",
       "      <td>147,3</td>\n",
       "      <td>264,12</td>\n",
       "      <td>0|1</td>\n",
       "      <td>26773694_G_C</td>\n",
       "      <td>26773694</td>\n",
       "      <td>57,242,1,11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr10</td>\n",
       "      <td>35633286</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype</td>\n",
       "      <td>SITE</td>\n",
       "      <td>14,18|1,1</td>\n",
       "      <td>34</td>\n",
       "      <td>...</td>\n",
       "      <td>14,2</td>\n",
       "      <td>0.182</td>\n",
       "      <td>16</td>\n",
       "      <td>2,1</td>\n",
       "      <td>6,0</td>\n",
       "      <td>8,1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>35633286_C_A</td>\n",
       "      <td>35633286</td>\n",
       "      <td>6,8,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr10</td>\n",
       "      <td>35633295</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype</td>\n",
       "      <td>SITE</td>\n",
       "      <td>13,19|1,1</td>\n",
       "      <td>34</td>\n",
       "      <td>...</td>\n",
       "      <td>15,2</td>\n",
       "      <td>0.167</td>\n",
       "      <td>17</td>\n",
       "      <td>3,0</td>\n",
       "      <td>6,0</td>\n",
       "      <td>9,1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>35633286_C_A</td>\n",
       "      <td>35633286</td>\n",
       "      <td>6,9,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr10</td>\n",
       "      <td>35633300</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype</td>\n",
       "      <td>SITE</td>\n",
       "      <td>14,21|1,1</td>\n",
       "      <td>37</td>\n",
       "      <td>...</td>\n",
       "      <td>18,2</td>\n",
       "      <td>0.143</td>\n",
       "      <td>20</td>\n",
       "      <td>4,1</td>\n",
       "      <td>7,0</td>\n",
       "      <td>11,1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>35633286_C_A</td>\n",
       "      <td>35633286</td>\n",
       "      <td>7,11,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr10</td>\n",
       "      <td>35633313</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events</td>\n",
       "      <td>SITE</td>\n",
       "      <td>15,25|2,2</td>\n",
       "      <td>46</td>\n",
       "      <td>...</td>\n",
       "      <td>22,4</td>\n",
       "      <td>0.168</td>\n",
       "      <td>26</td>\n",
       "      <td>8,1</td>\n",
       "      <td>6,1</td>\n",
       "      <td>14,2</td>\n",
       "      <td>8,14,2,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr10</td>\n",
       "      <td>87864482</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>85,87|2,2</td>\n",
       "      <td>182</td>\n",
       "      <td>...</td>\n",
       "      <td>77,2</td>\n",
       "      <td>0.042</td>\n",
       "      <td>79</td>\n",
       "      <td>21,0</td>\n",
       "      <td>23,1</td>\n",
       "      <td>44,1</td>\n",
       "      <td>38,39,1,1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr10</td>\n",
       "      <td>87952098</td>\n",
       "      <td>.</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact;slippage;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>78,94|7,6</td>\n",
       "      <td>188</td>\n",
       "      <td>...</td>\n",
       "      <td>82,7</td>\n",
       "      <td>0.116</td>\n",
       "      <td>89</td>\n",
       "      <td>16,3</td>\n",
       "      <td>32,3</td>\n",
       "      <td>49,6</td>\n",
       "      <td>38,44,4,3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr11</td>\n",
       "      <td>64809910</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>clustered_events;haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>143,30|1,4</td>\n",
       "      <td>205</td>\n",
       "      <td>...</td>\n",
       "      <td>70,4</td>\n",
       "      <td>0.054</td>\n",
       "      <td>74</td>\n",
       "      <td>38,2</td>\n",
       "      <td>27,0</td>\n",
       "      <td>65,3</td>\n",
       "      <td>0|1</td>\n",
       "      <td>64809910_G_A</td>\n",
       "      <td>64809910</td>\n",
       "      <td>57,13,1,3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr12</td>\n",
       "      <td>25245351</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>171,183|10,10</td>\n",
       "      <td>384</td>\n",
       "      <td>...</td>\n",
       "      <td>142,20</td>\n",
       "      <td>0.117</td>\n",
       "      <td>162</td>\n",
       "      <td>40,5</td>\n",
       "      <td>42,5</td>\n",
       "      <td>82,10</td>\n",
       "      <td>67,75,10,10</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7674242</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>100,103|2,2</td>\n",
       "      <td>207</td>\n",
       "      <td>...</td>\n",
       "      <td>70,4</td>\n",
       "      <td>0.070</td>\n",
       "      <td>74</td>\n",
       "      <td>10,0</td>\n",
       "      <td>29,2</td>\n",
       "      <td>39,2</td>\n",
       "      <td>37,33,2,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675088</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>108,110|4,5</td>\n",
       "      <td>230</td>\n",
       "      <td>...</td>\n",
       "      <td>85,9</td>\n",
       "      <td>0.112</td>\n",
       "      <td>94</td>\n",
       "      <td>31,4</td>\n",
       "      <td>27,2</td>\n",
       "      <td>58,7</td>\n",
       "      <td>42,43,4,5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675186</td>\n",
       "      <td>.</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>164,180|6,5</td>\n",
       "      <td>496</td>\n",
       "      <td>...</td>\n",
       "      <td>164,6</td>\n",
       "      <td>0.041</td>\n",
       "      <td>170</td>\n",
       "      <td>26,1</td>\n",
       "      <td>65,2</td>\n",
       "      <td>91,3</td>\n",
       "      <td>76,88,3,3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675966</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>29,24|1,2</td>\n",
       "      <td>62</td>\n",
       "      <td>...</td>\n",
       "      <td>22,3</td>\n",
       "      <td>0.150</td>\n",
       "      <td>25</td>\n",
       "      <td>7,0</td>\n",
       "      <td>5,1</td>\n",
       "      <td>13,2</td>\n",
       "      <td>0|1</td>\n",
       "      <td>7675966_T_A</td>\n",
       "      <td>7675966</td>\n",
       "      <td>12,10,1,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7675967</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>30,26|1,1</td>\n",
       "      <td>64</td>\n",
       "      <td>...</td>\n",
       "      <td>23,2</td>\n",
       "      <td>0.116</td>\n",
       "      <td>25</td>\n",
       "      <td>7,1</td>\n",
       "      <td>6,0</td>\n",
       "      <td>14,1</td>\n",
       "      <td>1|0</td>\n",
       "      <td>7675966_T_A</td>\n",
       "      <td>7675966</td>\n",
       "      <td>12,11,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr17</td>\n",
       "      <td>7676120</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>22,28|1,1</td>\n",
       "      <td>73</td>\n",
       "      <td>...</td>\n",
       "      <td>17,2</td>\n",
       "      <td>0.106</td>\n",
       "      <td>19</td>\n",
       "      <td>10,1</td>\n",
       "      <td>3,0</td>\n",
       "      <td>15,1</td>\n",
       "      <td>9,8,1,1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr18</td>\n",
       "      <td>51054859</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>149,152|9,6</td>\n",
       "      <td>323</td>\n",
       "      <td>...</td>\n",
       "      <td>127,15</td>\n",
       "      <td>0.131</td>\n",
       "      <td>142</td>\n",
       "      <td>21,6</td>\n",
       "      <td>51,4</td>\n",
       "      <td>72,10</td>\n",
       "      <td>59,68,9,6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr19</td>\n",
       "      <td>11021824</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype</td>\n",
       "      <td>SITE</td>\n",
       "      <td>27,27|1,1</td>\n",
       "      <td>56</td>\n",
       "      <td>...</td>\n",
       "      <td>18,2</td>\n",
       "      <td>0.167</td>\n",
       "      <td>20</td>\n",
       "      <td>4,0</td>\n",
       "      <td>5,1</td>\n",
       "      <td>9,1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>11021824_C_T</td>\n",
       "      <td>11021824</td>\n",
       "      <td>9,9,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr19</td>\n",
       "      <td>11021832</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>GCCTCCCTCTCCCGCTCTGGTCCTTCCT</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>27,26|1,1</td>\n",
       "      <td>56</td>\n",
       "      <td>...</td>\n",
       "      <td>18,2</td>\n",
       "      <td>0.167</td>\n",
       "      <td>20</td>\n",
       "      <td>4,0</td>\n",
       "      <td>5,1</td>\n",
       "      <td>9,1</td>\n",
       "      <td>0|1</td>\n",
       "      <td>11021824_C_T</td>\n",
       "      <td>11021824</td>\n",
       "      <td>9,9,1,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr19</td>\n",
       "      <td>15614428</td>\n",
       "      <td>.</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact;slippage;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>26,44|8,13</td>\n",
       "      <td>95</td>\n",
       "      <td>...</td>\n",
       "      <td>35,14</td>\n",
       "      <td>0.279</td>\n",
       "      <td>49</td>\n",
       "      <td>4,0</td>\n",
       "      <td>13,8</td>\n",
       "      <td>25,11</td>\n",
       "      <td>13,22,6,8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr22</td>\n",
       "      <td>20564559</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>49,87|2,4</td>\n",
       "      <td>145</td>\n",
       "      <td>...</td>\n",
       "      <td>76,3</td>\n",
       "      <td>0.053</td>\n",
       "      <td>79</td>\n",
       "      <td>12,0</td>\n",
       "      <td>40,1</td>\n",
       "      <td>52,2</td>\n",
       "      <td>32,44,1,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr3</td>\n",
       "      <td>41224670</td>\n",
       "      <td>.</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>55,58|2,2</td>\n",
       "      <td>121</td>\n",
       "      <td>...</td>\n",
       "      <td>40,4</td>\n",
       "      <td>0.119</td>\n",
       "      <td>44</td>\n",
       "      <td>4,0</td>\n",
       "      <td>17,2</td>\n",
       "      <td>21,2</td>\n",
       "      <td>20,20,2,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr3</td>\n",
       "      <td>172511139</td>\n",
       "      <td>.</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>4,15|1,2</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>8,3</td>\n",
       "      <td>0.300</td>\n",
       "      <td>11</td>\n",
       "      <td>4,1</td>\n",
       "      <td>2,1</td>\n",
       "      <td>6,2</td>\n",
       "      <td>2,6,1,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr3</td>\n",
       "      <td>179203819</td>\n",
       "      <td>.</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>53,52|1,2</td>\n",
       "      <td>114</td>\n",
       "      <td>...</td>\n",
       "      <td>50,3</td>\n",
       "      <td>0.097</td>\n",
       "      <td>53</td>\n",
       "      <td>2,0</td>\n",
       "      <td>25,2</td>\n",
       "      <td>27,2</td>\n",
       "      <td>0|1</td>\n",
       "      <td>179203819_TA_T</td>\n",
       "      <td>179203819</td>\n",
       "      <td>26,24,1,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr3</td>\n",
       "      <td>179203820</td>\n",
       "      <td>.</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>51,51|2,2</td>\n",
       "      <td>112</td>\n",
       "      <td>...</td>\n",
       "      <td>49,4</td>\n",
       "      <td>0.099</td>\n",
       "      <td>53</td>\n",
       "      <td>1,0</td>\n",
       "      <td>23,2</td>\n",
       "      <td>27,2</td>\n",
       "      <td>1|0</td>\n",
       "      <td>179203819_TA_T</td>\n",
       "      <td>179203819</td>\n",
       "      <td>25,24,2,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr5</td>\n",
       "      <td>80675082</td>\n",
       "      <td>.</td>\n",
       "      <td>AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTATTGGCATTGT...</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;normal_artifact;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>53,56|0,3</td>\n",
       "      <td>115</td>\n",
       "      <td>...</td>\n",
       "      <td>50,2</td>\n",
       "      <td>0.097</td>\n",
       "      <td>52</td>\n",
       "      <td>6,0</td>\n",
       "      <td>18,2</td>\n",
       "      <td>27,2</td>\n",
       "      <td>0|1</td>\n",
       "      <td>80675082_AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTAT...</td>\n",
       "      <td>80675082</td>\n",
       "      <td>23,27,0,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr5</td>\n",
       "      <td>80675095</td>\n",
       "      <td>.</td>\n",
       "      <td>CA</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;slippage;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>43,46|1,2</td>\n",
       "      <td>96</td>\n",
       "      <td>...</td>\n",
       "      <td>41,3</td>\n",
       "      <td>0.115</td>\n",
       "      <td>44</td>\n",
       "      <td>4,2</td>\n",
       "      <td>18,0</td>\n",
       "      <td>22,2</td>\n",
       "      <td>1|0</td>\n",
       "      <td>80675082_AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTAT...</td>\n",
       "      <td>80675082</td>\n",
       "      <td>19,22,1,2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chr6</td>\n",
       "      <td>43507679</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>18,37|1,2</td>\n",
       "      <td>62</td>\n",
       "      <td>...</td>\n",
       "      <td>26,3</td>\n",
       "      <td>0.131</td>\n",
       "      <td>29</td>\n",
       "      <td>0,0</td>\n",
       "      <td>8,1</td>\n",
       "      <td>19,2</td>\n",
       "      <td>7,19,1,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>chr6</td>\n",
       "      <td>156778847</td>\n",
       "      <td>.</td>\n",
       "      <td>GGGC</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact;slippage</td>\n",
       "      <td>SITE</td>\n",
       "      <td>90,85|4,5</td>\n",
       "      <td>196</td>\n",
       "      <td>...</td>\n",
       "      <td>82,5</td>\n",
       "      <td>0.070</td>\n",
       "      <td>87</td>\n",
       "      <td>40,2</td>\n",
       "      <td>12,1</td>\n",
       "      <td>52,3</td>\n",
       "      <td>45,37,2,3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>chr7</td>\n",
       "      <td>55181310</td>\n",
       "      <td>.</td>\n",
       "      <td>CA</td>\n",
       "      <td>TG</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>109,135|7,1</td>\n",
       "      <td>252</td>\n",
       "      <td>...</td>\n",
       "      <td>120,5</td>\n",
       "      <td>0.068</td>\n",
       "      <td>125</td>\n",
       "      <td>43,3</td>\n",
       "      <td>36,2</td>\n",
       "      <td>81,5</td>\n",
       "      <td>0|1</td>\n",
       "      <td>55181310_CA_TG</td>\n",
       "      <td>55181310</td>\n",
       "      <td>55,65,4,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chr7</td>\n",
       "      <td>55181313</td>\n",
       "      <td>.</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>haplotype;normal_artifact</td>\n",
       "      <td>SITE</td>\n",
       "      <td>109,135|7,1</td>\n",
       "      <td>254</td>\n",
       "      <td>...</td>\n",
       "      <td>120,5</td>\n",
       "      <td>0.068</td>\n",
       "      <td>125</td>\n",
       "      <td>43,3</td>\n",
       "      <td>38,2</td>\n",
       "      <td>81,5</td>\n",
       "      <td>0|1</td>\n",
       "      <td>55181310_CA_TG</td>\n",
       "      <td>55181310</td>\n",
       "      <td>55,65,4,1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>chr7</td>\n",
       "      <td>116771909</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>46,48|1,1</td>\n",
       "      <td>98</td>\n",
       "      <td>...</td>\n",
       "      <td>44,2</td>\n",
       "      <td>0.079</td>\n",
       "      <td>46</td>\n",
       "      <td>11,1</td>\n",
       "      <td>11,0</td>\n",
       "      <td>22,1</td>\n",
       "      <td>22,22,1,1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140778010</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>PASS</td>\n",
       "      <td>SITE</td>\n",
       "      <td>20,21|2,2</td>\n",
       "      <td>45</td>\n",
       "      <td>...</td>\n",
       "      <td>17,4</td>\n",
       "      <td>0.231</td>\n",
       "      <td>21</td>\n",
       "      <td>3,2</td>\n",
       "      <td>6,0</td>\n",
       "      <td>9,2</td>\n",
       "      <td>8,9,2,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr7</td>\n",
       "      <td>140781688</td>\n",
       "      <td>.</td>\n",
       "      <td>T</td>\n",
       "      <td>TCGAGTCCCGTCTACCAA</td>\n",
       "      <td>.</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>0,0|0,0</td>\n",
       "      <td>339</td>\n",
       "      <td>...</td>\n",
       "      <td>0,2</td>\n",
       "      <td>0.161</td>\n",
       "      <td>2</td>\n",
       "      <td>0,0</td>\n",
       "      <td>0,1</td>\n",
       "      <td>74,2</td>\n",
       "      <td>0,0,0,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chrX</td>\n",
       "      <td>20138666</td>\n",
       "      <td>.</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>normal_artifact;weak_evidence</td>\n",
       "      <td>weak_evidence</td>\n",
       "      <td>65,63|4,3</td>\n",
       "      <td>141</td>\n",
       "      <td>...</td>\n",
       "      <td>62,5</td>\n",
       "      <td>0.095</td>\n",
       "      <td>67</td>\n",
       "      <td>1,0</td>\n",
       "      <td>34,3</td>\n",
       "      <td>37,3</td>\n",
       "      <td>32,30,3,2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>36 rows √ó 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM        POS ID                                                REF  \\\n",
       "0    chr1   26773694  .                                                  G   \n",
       "1    chr1   26773703  .                                                  G   \n",
       "2    chr1   26773706  .                                                  G   \n",
       "3   chr10   35633286  .                                                  C   \n",
       "4   chr10   35633295  .                                                  A   \n",
       "5   chr10   35633300  .                                                  C   \n",
       "6   chr10   35633313  .                                                  A   \n",
       "7   chr10   87864482  .                                                  A   \n",
       "8   chr10   87952098  .                                                 CT   \n",
       "9   chr11   64809910  .                                                  G   \n",
       "12  chr12   25245351  .                                                  C   \n",
       "13  chr17    7674242  .                                                  A   \n",
       "14  chr17    7675088  .                                                  C   \n",
       "15  chr17    7675186  .                                                  A   \n",
       "16  chr17    7675966  .                                                  T   \n",
       "17  chr17    7675967  .                                                  G   \n",
       "18  chr17    7676120  .                                                  C   \n",
       "19  chr18   51054859  .                                                  C   \n",
       "20  chr19   11021824  .                                                  C   \n",
       "21  chr19   11021832  .                                                  G   \n",
       "22  chr19   15614428  .                                                 GA   \n",
       "23  chr22   20564559  .                                                  G   \n",
       "24   chr3   41224670  .                                                 AG   \n",
       "25   chr3  172511139  .                                                  G   \n",
       "26   chr3  179203819  .                                                 TA   \n",
       "27   chr3  179203820  .                                                 AT   \n",
       "28   chr5   80675082  .  AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTATTGGCATTGT...   \n",
       "29   chr5   80675095  .                                                 CA   \n",
       "30   chr6   43507679  .                                                  C   \n",
       "31   chr6  156778847  .                                               GGGC   \n",
       "32   chr7   55181310  .                                                 CA   \n",
       "33   chr7   55181313  .                                                  C   \n",
       "34   chr7  116771909  .                                                  T   \n",
       "35   chr7  140778010  .                                                  T   \n",
       "36   chr7  140781688  .                                                  T   \n",
       "37   chrX   20138666  .                                                 AT   \n",
       "\n",
       "                             ALT QUAL  \\\n",
       "0                              C    .   \n",
       "1                              A    .   \n",
       "2                              C    .   \n",
       "3                              A    .   \n",
       "4                              T    .   \n",
       "5                              G    .   \n",
       "6                              G    .   \n",
       "7                              G    .   \n",
       "8                              C    .   \n",
       "9                              A    .   \n",
       "12                             G    .   \n",
       "13                             G    .   \n",
       "14                             T    .   \n",
       "15                             G    .   \n",
       "16                             A    .   \n",
       "17                             T    .   \n",
       "18                             G    .   \n",
       "19                             G    .   \n",
       "20                             T    .   \n",
       "21  GCCTCCCTCTCCCGCTCTGGTCCTTCCT    .   \n",
       "22                             G    .   \n",
       "23                             A    .   \n",
       "24                             A    .   \n",
       "25                             A    .   \n",
       "26                             T    .   \n",
       "27                             A    .   \n",
       "28                             A    .   \n",
       "29                             C    .   \n",
       "30                             T    .   \n",
       "31                             G    .   \n",
       "32                            TG    .   \n",
       "33                             T    .   \n",
       "34                             G    .   \n",
       "35                             C    .   \n",
       "36            TCGAGTCCCGTCTACCAA    .   \n",
       "37                             A    .   \n",
       "\n",
       "                                        FILTER AS_FilterStatus    AS_SB_TABLE  \\\n",
       "0   clustered_events;haplotype;normal_artifact            SITE   104,453|1,21   \n",
       "1   clustered_events;haplotype;normal_artifact            SITE   131,481|1,21   \n",
       "2   clustered_events;haplotype;normal_artifact            SITE   133,481|1,21   \n",
       "3                   clustered_events;haplotype            SITE      14,18|1,1   \n",
       "4                   clustered_events;haplotype            SITE      13,19|1,1   \n",
       "5                   clustered_events;haplotype            SITE      14,21|1,1   \n",
       "6                             clustered_events            SITE      15,25|2,2   \n",
       "7                              normal_artifact            SITE      85,87|2,2   \n",
       "8       normal_artifact;slippage;weak_evidence   weak_evidence      78,94|7,6   \n",
       "9   clustered_events;haplotype;normal_artifact            SITE     143,30|1,4   \n",
       "12                                        PASS            SITE  171,183|10,10   \n",
       "13                                        PASS            SITE    100,103|2,2   \n",
       "14                                        PASS            SITE    108,110|4,5   \n",
       "15                             normal_artifact            SITE    164,180|6,5   \n",
       "16                                        PASS            SITE      29,24|1,2   \n",
       "17                                        PASS            SITE      30,26|1,1   \n",
       "18                                        PASS            SITE      22,28|1,1   \n",
       "19                                        PASS            SITE    149,152|9,6   \n",
       "20                                   haplotype            SITE      27,27|1,1   \n",
       "21                     haplotype;weak_evidence   weak_evidence      27,26|1,1   \n",
       "22      normal_artifact;slippage;weak_evidence   weak_evidence     26,44|8,13   \n",
       "23                             normal_artifact            SITE      49,87|2,4   \n",
       "24                               weak_evidence   weak_evidence      55,58|2,2   \n",
       "25                                        PASS            SITE       4,15|1,2   \n",
       "26                     haplotype;weak_evidence   weak_evidence      53,52|1,2   \n",
       "27                     haplotype;weak_evidence   weak_evidence      51,51|2,2   \n",
       "28     haplotype;normal_artifact;weak_evidence   weak_evidence      53,56|0,3   \n",
       "29            haplotype;slippage;weak_evidence   weak_evidence      43,46|1,2   \n",
       "30                                        PASS            SITE      18,37|1,2   \n",
       "31                    normal_artifact;slippage            SITE      90,85|4,5   \n",
       "32                   haplotype;normal_artifact            SITE    109,135|7,1   \n",
       "33                   haplotype;normal_artifact            SITE    109,135|7,1   \n",
       "34                                        PASS            SITE      46,48|1,1   \n",
       "35                                        PASS            SITE      20,21|2,2   \n",
       "36                               weak_evidence   weak_evidence        0,0|0,0   \n",
       "37               normal_artifact;weak_evidence   weak_evidence      65,63|4,3   \n",
       "\n",
       "     DP  ... tumor_AD tumor_AF tumor_DP tumor_F1R2 tumor_F2R1 tumor_FAD  \\\n",
       "0   581  ...   276,12    0.050      288       97,8      137,3    245,12   \n",
       "1   634  ...   299,12    0.047      311      115,9      146,2    264,12   \n",
       "2   644  ...   299,12    0.047      311      114,9      147,3    264,12   \n",
       "3    34  ...     14,2    0.182       16        2,1        6,0       8,1   \n",
       "4    34  ...     15,2    0.167       17        3,0        6,0       9,1   \n",
       "5    37  ...     18,2    0.143       20        4,1        7,0      11,1   \n",
       "6    46  ...     22,4    0.168       26        8,1        6,1      14,2   \n",
       "7   182  ...     77,2    0.042       79       21,0       23,1      44,1   \n",
       "8   188  ...     82,7    0.116       89       16,3       32,3      49,6   \n",
       "9   205  ...     70,4    0.054       74       38,2       27,0      65,3   \n",
       "12  384  ...   142,20    0.117      162       40,5       42,5     82,10   \n",
       "13  207  ...     70,4    0.070       74       10,0       29,2      39,2   \n",
       "14  230  ...     85,9    0.112       94       31,4       27,2      58,7   \n",
       "15  496  ...    164,6    0.041      170       26,1       65,2      91,3   \n",
       "16   62  ...     22,3    0.150       25        7,0        5,1      13,2   \n",
       "17   64  ...     23,2    0.116       25        7,1        6,0      14,1   \n",
       "18   73  ...     17,2    0.106       19       10,1        3,0      15,1   \n",
       "19  323  ...   127,15    0.131      142       21,6       51,4     72,10   \n",
       "20   56  ...     18,2    0.167       20        4,0        5,1       9,1   \n",
       "21   56  ...     18,2    0.167       20        4,0        5,1       9,1   \n",
       "22   95  ...    35,14    0.279       49        4,0       13,8     25,11   \n",
       "23  145  ...     76,3    0.053       79       12,0       40,1      52,2   \n",
       "24  121  ...     40,4    0.119       44        4,0       17,2      21,2   \n",
       "25   22  ...      8,3    0.300       11        4,1        2,1       6,2   \n",
       "26  114  ...     50,3    0.097       53        2,0       25,2      27,2   \n",
       "27  112  ...     49,4    0.099       53        1,0       23,2      27,2   \n",
       "28  115  ...     50,2    0.097       52        6,0       18,2      27,2   \n",
       "29   96  ...     41,3    0.115       44        4,2       18,0      22,2   \n",
       "30   62  ...     26,3    0.131       29        0,0        8,1      19,2   \n",
       "31  196  ...     82,5    0.070       87       40,2       12,1      52,3   \n",
       "32  252  ...    120,5    0.068      125       43,3       36,2      81,5   \n",
       "33  254  ...    120,5    0.068      125       43,3       38,2      81,5   \n",
       "34   98  ...     44,2    0.079       46       11,1       11,0      22,1   \n",
       "35   45  ...     17,4    0.231       21        3,2        6,0       9,2   \n",
       "36  339  ...      0,2    0.161        2        0,0        0,1      74,2   \n",
       "37  141  ...     62,5    0.095       67        1,0       34,3      37,3   \n",
       "\n",
       "      tumor_PGT                                          tumor_PID   tumor_PS  \\\n",
       "0           0|1                                       26773694_G_C   26773694   \n",
       "1           0|1                                       26773694_G_C   26773694   \n",
       "2           0|1                                       26773694_G_C   26773694   \n",
       "3           0|1                                       35633286_C_A   35633286   \n",
       "4           0|1                                       35633286_C_A   35633286   \n",
       "5           0|1                                       35633286_C_A   35633286   \n",
       "6      8,14,2,2                                               None       None   \n",
       "7     38,39,1,1                                               None       None   \n",
       "8     38,44,4,3                                               None       None   \n",
       "9           0|1                                       64809910_G_A   64809910   \n",
       "12  67,75,10,10                                               None       None   \n",
       "13    37,33,2,2                                               None       None   \n",
       "14    42,43,4,5                                               None       None   \n",
       "15    76,88,3,3                                               None       None   \n",
       "16          0|1                                        7675966_T_A    7675966   \n",
       "17          1|0                                        7675966_T_A    7675966   \n",
       "18      9,8,1,1                                               None       None   \n",
       "19    59,68,9,6                                               None       None   \n",
       "20          0|1                                       11021824_C_T   11021824   \n",
       "21          0|1                                       11021824_C_T   11021824   \n",
       "22    13,22,6,8                                               None       None   \n",
       "23    32,44,1,2                                               None       None   \n",
       "24    20,20,2,2                                               None       None   \n",
       "25      2,6,1,2                                               None       None   \n",
       "26          0|1                                     179203819_TA_T  179203819   \n",
       "27          1|0                                     179203819_TA_T  179203819   \n",
       "28          0|1  80675082_AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTAT...   80675082   \n",
       "29          1|0  80675082_AAAATGTTAGGGACAAAAAAAAGGGCAACATTTTTAT...   80675082   \n",
       "30     7,19,1,2                                               None       None   \n",
       "31    45,37,2,3                                               None       None   \n",
       "32          0|1                                     55181310_CA_TG   55181310   \n",
       "33          0|1                                     55181310_CA_TG   55181310   \n",
       "34    22,22,1,1                                               None       None   \n",
       "35      8,9,2,2                                               None       None   \n",
       "36      0,0,0,2                                               None       None   \n",
       "37    32,30,3,2                                               None       None   \n",
       "\n",
       "       tumor_SB  \n",
       "0   45,231,1,11  \n",
       "1   57,242,1,11  \n",
       "2   57,242,1,11  \n",
       "3       6,8,1,1  \n",
       "4       6,9,1,1  \n",
       "5      7,11,1,1  \n",
       "6          None  \n",
       "7          None  \n",
       "8          None  \n",
       "9     57,13,1,3  \n",
       "12         None  \n",
       "13         None  \n",
       "14         None  \n",
       "15         None  \n",
       "16    12,10,1,2  \n",
       "17    12,11,1,1  \n",
       "18         None  \n",
       "19         None  \n",
       "20      9,9,1,1  \n",
       "21      9,9,1,1  \n",
       "22         None  \n",
       "23         None  \n",
       "24         None  \n",
       "25         None  \n",
       "26    26,24,1,2  \n",
       "27    25,24,2,2  \n",
       "28    23,27,0,2  \n",
       "29    19,22,1,2  \n",
       "30         None  \n",
       "31         None  \n",
       "32    55,65,4,1  \n",
       "33    55,65,4,1  \n",
       "34         None  \n",
       "35         None  \n",
       "36         None  \n",
       "37         None  \n",
       "\n",
       "[36 rows x 47 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_tumor_variants = df[df['tumor_AF'] > normal_af]\n",
    "filtered_tumor_variants.to_csv(f'{RESULTS_FOLDER}/filtered_tumor_variants.csv')\n",
    "filtered_tumor_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d3fe29b3-f22c-493e-9b11-05b5dfa5fac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['.'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['ID'].unique()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
